### ISSUE TITLE
bug: certain query strings on `MockHtmlElement` crash due to `getAttributeNode` not being defined

### ISSUE THREAD

**Issue by Retsam** [ionic-team/stencil#3043]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/master/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/master/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.6.0

### Current Behavior

With `mock-doc`'s `MockHtmlElement`, passing specific query strings to `.querySelector()` causes the method to crash due to a missing implementation for [`Element.getAttributeNode`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttributeNode).  

The simplest reproduction I could find is `mockEl.querySelector('audio[controls],video[controls]')`.   (The original bug comes form the [tabbable](https://github.com/focus-trap/tabbable/blob/master/src/index.js#L14) library's query-selector to find safely tabbable elements)

### Expected Behavior

The query does not crash (and ideally returns the correct elements).

### Steps to Reproduce

Since it's an issue with the mock element used in node tests, this can be reproduced via node:
```js
require('@stencil/core/mock-doc')
    .createDocument()
    .documentElement
    .querySelector('audio[controls],video[controls]');
```
(As a single command)
```sh
node -e "require('@stencil/core/mock-doc').createDocument().documentElement.querySelector('audio[controls],video[controls]');"
```

### Code Reproduction URL

https://gist.github.com/Retsam/08df2a1e8c5f555221af47251b20e3cc

### Additional Information

A simple `() => null` polyfill might suffice; it's likely what we'll be doing to get our tests to not crash in the meantime.

**Comment by splitinfinities** [comment]
note: we may need to add further mocks to /stencil/src/mock-doc/element.ts for audio and video.

**Comment by ianvieira** [comment]
Any update on this issue? It's almost an year old and it makes very tricky to use `querySelectors` on the components, if we cannot test them

**Comment by angelo-v** [comment]
This error also occurs when using testing-library to test stencil. Some queries like `getByRole` are affected.

**Comment by alicewriteswrongs** [comment]
The fix for this issue has been released as a part of today's [Stencil v4.8.0 release](https://github.com/ionic-team/stencil/releases/tag/v4.8.0)!

**Comment by angelo-v** [comment]
> This error also occurs when using testing-library to test stencil. Some queries like `getByRole` are affected.

While this specific error is gone, there is now another one for `getByRole` https://github.com/ionic-team/stencil/issues/5252
